src/app/qgshandlebadlayers.cpp

```diff
@@ -383,12 +386,23 @@ void QgsHandleBadLayers::apply()
     QTableWidgetItem *item = mLayerList->item( i, 4 );
     QString datasource = item->text();
     QString fileName;
+    QString formatter;
     const QString layerId { node.namedItem( QStringLiteral( "id" ) ).toElement().text() };
     const QString name { mLayerList->item( i, 0 )->text() };
+    QString provider = node.namedItem( QStringLiteral( "provider" ) ).toElement().text();
+    if ( provider == QString( "spatialite" ) )
+    {
+        if ( datasource.contains( "dbname='" ) )
```

Comment:
Could the provider decodeUri api be used here instead of the raw string handling?

Reply:
In my test I had bad results with it but give me a moment I'll try something else.

Reply:
reworked the uri handling to use the raw string instead of an altered string, other behavior seems unchanged.